asreview/project.py

```diff
@@ -107,25 +111,47 @@ def get_projects(project_paths=None):
     return [ASReviewProject(project_path) for project_path in project_paths]
 
 
-def _create_project_id(name):
-    """Create project id from input name."""
+def _create_plaintext_project_id(title):
+    """Create plaintext project id from input title."""
 
-    if isinstance(name, str) and len(name) > 0 and not name[0].isalnum():
+    if isinstance(title, str) and len(title) > 0 and not title[0].isalnum():
         raise ValueError(
             "First character should be alphabet" " letter (a-z) or number (0-9)."
         )
 
-    if not name and not isinstance(name, str) and len(name) >= 3:
-        raise ValueError("Project name should be at least 3 characters.")
+    if not title and not isinstance(title, str) and len(title) >= 3:
+        raise ValueError("Project title should be at least 3 characters.")
 
-    project_id = ""
-    for c in name.lower():
+    plaintext_project_id = ""
+    for c in title.lower():
         if c.isalnum():
-            project_id += c
-        elif len(project_id) > 0 and project_id[-1] != "-":
-            project_id += "-"
+            plaintext_project_id += c
+        elif len(plaintext_project_id) > 0 and plaintext_project_id[-1] != "-":
+            plaintext_project_id += "-"
 
-    return project_id
+    return plaintext_project_id
+
+
+def _get_user_uuid(user_id=""):
+    user_uuid = uuid5(NAMESPACE_URL, str(user_id)).hex
+    return user_uuid
+
+
+def _get_project_uuid(project_title, user):
+    """Get the project uuid from project title and user id."""
+
+    plaintext_project_id = _create_plaintext_project_id(project_title)
+
+    # Get the user uuid.
+    if app_is_authenticated(current_app):
+        user_uuid = _get_user_uuid(user.id)
+    else:
+        user_uuid = _get_user_uuid()
+
+    # Get the project uuid.
+    project_uuid = uuid5(NAMESPACE_URL, plaintext_project_id + user_uuid).hex
+
+    return project_uuid
```

Comment:
Can't we replace this by `uuid4().hex`? The title, namespace and username can be left out isn't it? That solves your issue with temporary projects.